ports/unix/mpthreadport.c

```diff
@@ -271,6 +276,36 @@ int mp_thread_mutex_lock(mp_thread_mutex_t *mutex, int wait) {
     return -ret;
 }
 
+#ifdef MICROPY_PY_THREAD_TIMEDLOCK
+int mp_thread_mutex_lock_timed(mp_thread_mutex_t *mutex, mp_float_t timeout) {
+    int ret;
+    struct timeval _timeval;
+    struct timezone _timezone;
+    gettimeofday(&_timeval, &_timezone);
+    uint32_t _timeout_nano = ((uint32_t) (1e9*timeout)) + 1000 * _timeval.tv_usec;
```

Comment:
Maybe do the float conversion like `ports/unix/modtime.c`?

```C
    struct timeval tv;
    mp_float_t val = mp_obj_get_float(arg);
    double ipart;
    tv.tv_usec = round(modf(val, &ipart) * 1000000);
    tv.tv_sec = ipart;
```

Otherwise, `_timeout_nano` probably needs to be `uint64_t`

Reply:
In the updated version I am handling the `float` to `int` conversion in `modthread.c` and passing an int number of milliseconds to `mp_thread_mutex_lock_timeout`. See my comment regarding whether one should support `int` *and* `float` as timeout argument.

Regarding the overflow issue, the updated version is safe with `uint32_t` because `_timeout_nano` will always be smaller than 2 seconds.